Declare DEV mode in tests using conftest.py

https://github.com/spyder-ide/spyder-terminal/blob/1797a86ee0f4e94627493b93440b5b99f15437a5/spyder_terminal/terminalplugin.py#L76-L80

https://github.com/spyder-ide/spyder-terminal/blob/1797a86ee0f4e94627493b93440b5b99f15437a5/spyder_terminal/tests/test_terminal.py#L183-L190


The terminal is set up by the fixture `setup_terminal` **before** the `DEV` flag is set manually here. What's the point on that?

Fails on packaging for OpenSUSE because DEV mode is not enabled by default:
```
[   41s] =================================== FAILURES ===================================
[   41s] ___________________________ test_output_redirection ____________________________
[   41s] 
[   41s] setup_terminal = <spyder_terminal.terminalplugin.TerminalPlugin object at 0x7f96cd743280>
[   41s] qtbot_module = <pytestqt.qtbot.QtBot object at 0x7f96cd6d6fa0>
[   41s] 
[   41s]     def test_output_redirection(setup_terminal, qtbot_module):
[   41s]         """Test if stdout and stderr are redirected on DEV mode."""
[   41s]         spyder_terminal.terminalplugin.DEV = True
[   41s]     
[   41s]         stdout = osp.join(getcwd(), 'spyder_terminal_out.log')
[   41s]         stderr = osp.join(getcwd(), 'spyder_terminal_err.log')
[   41s] >       assert osp.exists(stdout) and osp.exists(stderr)
[   41s] E       AssertionError: assert (False)
[   41s] E        +  where False = <function exists at 0x7f96e9e2fe50>('/home/abuild/rpmbuild/BUILD/spyder-terminal-0.3.0/spyder_terminal_out.log')
[   41s] E        +    where <function exists at 0x7f96e9e2fe50> = osp.exists
[   41s] 
[   41s] spyder_terminal/tests/test_terminal.py:189: AssertionError
```

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -22,7 +22,6 @@ jobs:
     environment:
       - PYTHON_VERSION: "3.6"
       - USE_CONDA: "yes"
-      - SPYDER_DEV: true
       - CODECOV_TOKEN: "273c981b-7851-4895-a4bb-07377c67791e"
 
   python3.7:
@@ -30,7 +29,6 @@ jobs:
     environment:
       - PYTHON_VERSION: "3.7"
       - USE_CONDA: "yes"
-      - SPYDER_DEV: true
       - CODECOV_TOKEN: "273c981b-7851-4895-a4bb-07377c67791e"
 
 workflows:

--- a/azure-pipelines.yml
+++ b/azure-pipelines.yml
@@ -16,7 +16,6 @@ jobs:
     USE_CONDA: "yes"
     CI: True
     AZURE: True
-    SPYDER_DEV: True
 
   # Run the pipeline with multiple Python versions
   strategy:
@@ -95,7 +94,6 @@ jobs:
     AZURE: True
     PYTHON: "C:\\Miniconda"
     PYWINPTY_BLOCK: 0
-    SPYDER_DEV: True
     CODECOV_TOKEN: "273c981b-7851-4895-a4bb-07377c67791e"
 
   # Run the pipeline with multiple Python versions

--- a/conftest.py
+++ b/conftest.py
@@ -1,0 +1,15 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright © Spyder Project Contributors
+# Licensed under the terms of the MIT License
+#
+
+"""
+Configuration file for Pytest
+NOTE: DO NOT add fixtures here. It could generate problems with
+      QtAwesome being called before a QApplication is created.
+"""
+
+import os
+
+os.environ['SPYDER_DEV'] = 'True'

